components/browser/menu/src/main/java/mozilla/components/browser/menu/BrowserMenu.kt

```diff
@@ -22,7 +23,8 @@ import mozilla.components.support.ktx.android.view.isRTL
  * A popup menu composed of BrowserMenuItem objects.
  */
 class BrowserMenu internal constructor(
-    private val adapter: BrowserMenuAdapter
+    private val adapter: BrowserMenuAdapter,
+    private val cardBackgroundColor: Int? = null
```

Comment:
Maybe we can do this differently by retrieving the background colour from an attribute instead of having it passed through the constructor.

Consider what would happen if we wanted to add more styling attributes, do we continue adding more constructor args? Is a background colour really needed for constructing a BrowserMenu?

There is an `init` block in [BrowserToolbar.kt](https://github.com/mozilla-mobile/android-components/blob/master/components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/BrowserToolbar.kt#L275-L320) that gets styled attributes from XML which might be an option. Although, I don't think you need the Kotlin API for this that class does.

Reply:
I agree that it would be nice if we could reduce the number of parameters here. I don't think we can use XML attributes though: The menu is created by the builder and never placed in XML. Having those attributes on the toolbar may not make sense if the menu component can be used independently. Even if we add it to the toolbar: We would still need to pass it to the builder and/or menu.

So I wonder if this could be a great candidate for a style that can be overridden in the app theme though.

(nit: Let's not call it `cardBackgroundColor` since the usage of `CardView` is an implementation detail here)

Reply:
This looks like a great article describing the approach:
https://blog.danlew.net/2011/03/03/custom_styles_in_android_libraries/

Reply:
Just noticed that the menu already has a bunch of styles :)

https://github.com/mozilla-mobile/android-components/blob/master/components/browser/menu/src/main/res/values/style.xml

Reply:
My initial thought was the same as yours, @jonalmeida but as @pocmo mentioned the "menu" was never sticking around to be modified (simply created by the builder). I will take a look at the style file you linked, Sebastian.

Thanks for all the great feedback everyone 😄